### DISCUSSION TITLE
loadingRow event not firing

### DISCUSSION THREAD

**Discussion by ollih1980** [AvaloniaUI/Avalonia#7186] [Q&A]
Hello Community,

When I try to set the Window Height to e.g. 0 and then back to e.g. 500 the loadingRow event in my DataGrid is not firing.
But when I do it manually, so resizing the windows via hand, it's firing.

Any idea how to archive this?

**Comment by maxkatz6** [comment]
I guess it's just means that some rows vere loaded for DataGrid with 0 height and nothing new was loaded after changing height to 500. Have you checked with dev tools. It's easy to see how many rows are loaded to the memory therr.

**Reply by ollih1980** [reply]
I'm trying to force the loadingRow event to fire. Therefore I'm setting the MainWindows size to 0 and directly after this again to e.g. 500.
When I do it manually, the event is firing. When I do this via code, it's not firing.

```
this.Heigth = 0;
this.Heigth = 500;
```
Which parameter is the "how many rows are loaded to the memory" one? I'm not finding anything like this.

Thx.

**Reply by maxkatz6** [reply]
> I'm trying to force the loadingRow event to fire

But why?

> Which parameter is the "how many rows are loaded to the memory" one? I'm not finding anything like this.

As minimum as possible. I don't think you can configure it.

**Reply by ollih1980** [reply]
Why? :
I'm trying to change the row color within the DataGrid with a click on a button.  And to do this I need to handle it via loadingRow, as Triggers are not yet supported like in WPF.

**Reply by maxkatz6** [reply]
You don't need thiggers for that. 
First of all you need to have bindable property in data context. And create binding to it in LoadingRow method. 
https://docs.avaloniaui.net/docs/data-binding/binding-from-code
You can go with INotifyPropertyChanged approach (or any other if you want), just update property on button click. 

You are using framework that lives with bindings after all. With converter on top of it you can change the color. 
And ideally you should dispose this binding in UnloadingRow, but it will work without it.

**Reply by ollih1980** [reply]
Great, that works! Thx a lot!
